controllers/pkg/reconcilers/repository/README.md

```diff
@@ -0,0 +1,39 @@
+# repository controller
+
+The repo controller is a k8s controller acting on repository.infra.nephio.org and handles the lifecycle of the repository in gitea.
```

Comment:
Post R1, we should consider separating consumption/requirements/provisioning resources.

Reply:
agreed we should reflect on the purpose of these and define the right strategy behind each one.